statistics/update.go

```diff
@@ -143,6 +144,68 @@ func (h *Handle) dumpTableStatDeltaToKV(id int64, delta variable.TableDelta) err
 	return errors.Trace(err)
 }
 
+// QueryFeedback is used to represent the query stats info.
+type QueryFeedback struct {
+	tableID     int64
+	colID       int64
+	intRanges   []ranger.IntColumnRange
+	idxRanges   []*ranger.IndexRange
+	histVersion uint64 // histVersion is the version of the histogram when we issue the query.
+	expected    int64
+	actual      int64
+}
+
+// NewQueryFeedback returns a new query feedback.
+func NewQueryFeedback(tableID int64, colID int64, histVer uint64, expected int64) *QueryFeedback {
+	return &QueryFeedback{
+		tableID:     tableID,
+		colID:       colID,
+		histVersion: histVer,
+		expected:    expected,
+		actual:      0}
+}
+
+// Increase increases the actual count. If there are some errors happened, then `count` will be less than 0.
+func (q *QueryFeedback) Increase(count int64) {
```

Comment:
Is it possible that the actual is set to -1, then later increased to a positive count?